src/session.rs

```diff
@@ -83,21 +88,54 @@ impl SessionWrapper {
         &self,
         query: &PreparedStatementWrapper,
         params: Vec<Option<&QueryParameterWrapper>>,
-        _options: &QueryParameterWrapper,
+        options: &QueryOptionsWrapper,
     ) -> napi::Result<QueryResultWrapper> {
         let params_vec: Vec<Option<CqlValue>> = params
             .iter()
             .map(|e| e.as_ref().map(|v| v.parameter.clone()))
             .collect();
+        let query = apply_options(query.prepared.clone(), options)?;
         Ok(QueryResultWrapper::from_query(
             self.internal
-                .execute_unpaged(&query.prepared, params_vec)
+                .execute_unpaged(&query, params_vec)
                 .await
                 .map_err(err_to_napi)?,
         ))
     }
 }
 
+fn apply_options(
+    mut query: PreparedStatement,
+    options: &QueryOptionsWrapper,
+) -> napi::Result<PreparedStatement> {
+    if let Some(o) = options.consistency {
+        query.set_consistency(
+            Consistency::try_from(o)
+                .map_err(|_| js_error(format!("Unknown consistency value: {o}")))?,
+        );
+    }
+    if let Some(o) = options.is_idempotent {
+        query.set_is_idempotent(o);
+    }
+    if let Some(o) = options.fetch_size {
+        if o.is_negative() {
+            return Err(js_error("fetch size cannot be negative"));
+        }
+        query.set_page_size(o);
+    }
```

Comment:
One important note about `fetch_size`:

- In Rust driver, we used to have `page_size` as `Option<i32>`. `None` meant no paging, and `Some(n)` would use pages of size `n`. This approach showed to be error-prone, so we changed it in a way that the page size is always some positive number, and it's the execution method that chooses between paged and unpaged queries (`query_unpaged` vs `query_single_page` and `query_iter`).
- in JS driver, what was true for Rust driver in the past may still be true, i.e. positive page size might imply paging and `0` page size - no paging.

Please make sure to support those semantics properly. Not to forget about it, I suggest opening an issue for that.